modules/tracktion_engine/timestretch/tracktion_RubberBand.cpp

```diff
@@ -0,0 +1,365 @@
+{
+    // Stretches the audio by /2, at the same pitch
+    setSpeedAndPitch(0.5f, 1.0f);
+
+    AudioBuffer<float> sinBuffer;
+    sinBuffer.setSize(2, 44100);
+
+    auto wL = sinBuffer.getWritePointer(0);
+    auto wR = sinBuffer.getWritePointer(1);
+
+    double currentAngle = 0.0, angleDelta = 0.0;
+    float originalPitch = 432.0f; //A4
+    auto cyclesPerSample = originalPitch / 44100.0f;
+    angleDelta = cyclesPerSample * 2.0 * juce::MathConstants<double>::pi;
+
+    // Generate sin wave and store in testBuffer
+    for (int sample = 0; sample < 44100; ++sample)
+    {
+        auto currentSample = (float)std::sin(currentAngle);
+        currentAngle += angleDelta;
+        wL[sample] = currentSample;
+        wR[sample] = currentSample;
+    }
+
+    // Process sin wave to time stretch and store in testBuffer
+    processData(sinBuffer.getArrayOfReadPointers(), 44100, sinBuffer.getArrayOfWritePointers());
+
+
+    AudioBuffer<float> testBuffer;
+    testBuffer.setSize(2, 22050);
+
+    auto t = testBuffer.getWritePointer(0);
+    auto s = sinBuffer.getWritePointer(0);
+
+    int scount = 0;
+
+    for (int sample = 0; sample < 44100; ++sample)
+    {
+        if (s[sample] != 0 && sample >= 22050)
+        {
+            t[sample] = s[sample];
+        }
+        else
+        {
+            scount++;
+        }
+    }
+
+
+
+    // Check whether buffer has resized
+    if (abs(sinBuffer.getNumSamples() - testBuffer.getNumSamples() * 2.0f) < tolerance && scount == 22050)
+        return true;
+
+    return false;
+}
+
+float RubberStretcher::getPitchFromNumZeroCrossings(int numZeroCrossings, int numSamples)
+{
+    float bufferTimeInSeconds = numSamples / sampleRate;
+
+    int numCycles = numZeroCrossings / 2;
+
+    float pitchInHertz = numCycles / bufferTimeInSeconds;
+
+    return pitchInHertz;
+}
+
+int RubberStretcher::getNumZeroCrossings(AudioBuffer<float> buffer)
+{
+    auto dest = buffer.getWritePointer(0);
+
+    int numZeroCrossings = 0;
+
+    for (int sample = 1; sample < buffer.getNumSamples(); ++sample)
+    {
+        if ((dest[sample] < 0.0f && dest[sample - 1] > 0.0f)
+            || (dest[sample] > 0.0f && dest[sample - 1] < 0.0f))
+        {
+            numZeroCrossings++;
+        }
+    }
+
+    return numZeroCrossings;
+}
+
+
+//==============================================================================
+//==============================================================================
+class RubberBandStretcherTests  : public UnitTest
+{
+public:
+    RubberBandStretcherTests()
+        : UnitTest ("RubberBandStretcherTests", "Tracktion")
+    {
+    }
+
+    void runTest() override
+    {
+        runPitchShiftTest();
+        runTimeStretchTest();
+    }
+
+private:
+    void runPitchShiftTest()
+    {
+        beginTest ("Pitch shift");
+        
+        RubberStretcher thisRubberBandStretcher = RubberStretcher (44100, 512, 2, RubberBandStretcher::Option::OptionProcessRealTime);
```

Comment:
This declaration has some redundant repetition. It can simply be written as `RubberStretcher thisRubberBandStretcher (44100, 512, 2, RubberBandStretcher::Option::OptionProcessRealTime);`

It's also probably better to use a constant variable for the sample rate, buffer size and num channels are they're used several times in the method.